ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -140,6 +140,19 @@ def _check_image_exists(self):
                 raise EC2UploadImgException(msg)
 
     # ---------------------------------------------------------------------
+    def _check_virt_type_consistent(self):
+        """When using root swap the virtualization type of the helper
+           image and the target image must be the same"""
+        image = self._connect().describe_images(
```

Comment:
I *really* prefer the resource objects over the raw client description dicts ;-)
```
image = boto3.resource('ec2').Image(self.launch_ami_id)
if not self.image_virt_type == image.image.virtualization_type
    ...
```